var/spack/repos/builtin/packages/hdf/package.py

```diff
@@ -15,45 +15,121 @@ class Hdf(AutotoolsPackage):
     list_url = "https://support.hdfgroup.org/ftp/HDF/releases/"
     list_depth = 2
 
+    version('4.2.15', sha256='dbeeef525af7c2d01539906c28953f0fdab7dba603d1bc1ec4a5af60d002c459')
     version('4.2.14', sha256='2d383e87c8a0ca6a5352adbd1d5546e6cc43dc21ff7d90f93efa644d85c0b14a')
     version('4.2.13', sha256='be9813c1dc3712c2df977d4960e1f13f20f447dfa8c3ce53331d610c1f470483')
     version('4.2.12', sha256='dd419c55e85d1a0e13f3ea5ed35d00710033ccb16c85df088eb7925d486e040c')
     version('4.2.11', sha256='c3f7753b2fb9b27d09eced4d2164605f111f270c9a60b37a578f7de02de86d24')
 
     variant('szip', default=False, description="Enable szip support")
-    variant('libtirpc', default=False, description="Use xdr library from libtirpc package; if false, will use system or hdf internal")
+    variant('xdr', default='system', values=('system', 'embedded', 'external'),
```

Comment:
System means pick up the library from the OS? We almost never want that, as it leads to unreproducible builds and makes building a binary cache a problem. I would remove that option altogether.

Reply:
@adamjstewart `system` actually meant "use xdr implementation from `libc`". However, I have changed this. Now there is only a boolean variant `external-xdr`. If it's `True`, the package depends on a virtual package `rpc`, which is provided by `libtirpc` and `libc`. Otherwise, an internal implementation for `xdr` is used.